yktracec/src/jitmodbuilder.cc

```diff
@@ -1811,10 +1752,12 @@ class JITModBuilder {
                 assert(cast<GetElementPtrInst>(I)->getPointerOperand() ==
                        TraceInputs);
                 I++;
+                CurInstrIdx++;
```

Comment:
Are `I` and `CurInstrIdx` now effectively the same value -- they seem to be always incremented together?

Reply:
I is an instruction iterator though.

Reply:
`I` is an iterator pointing to an `Instruction`. `CurInstrIdx` is an integer referring to the index of that particular instruction in its parent basic block.

Reply:
Does the iterator / instruction know its index?

Reply:
AFAIK, it does not :)

Reply:
I have a vague memory (maybe fabricated) that there is an iterator adaptor for enumerating iterators, but we'd have to be careful to always add an offset since we don't always start from instruction index 0 (due to resume points).

In fact, I think it would make comprehension harder, so let's not do that.